### PR TITLE
M2-4835: Linting rules for `mindlogger-web-refactor`

### PR DESCRIPTION
**Pull request by sultanofcardio**
resolves: [M2-4835](https://mindlogger.atlassian.net/browse/M2-4835)

# Objective

This PR creates a GitHub Action to run linting tasks on pull requests submitted to the repository. As part of the ticket, I took the opportunity to update the ESlint rules on the project to be more in line with those on [mindlogger-admin](https://github.com/ChildMindInstitute/mindlogger-admin). These updates have raised a number of linting issues, but **won't block** the merging of pull requests. While I've addressed some of the smaller items (and any error ones), I expect that these will be addressed in the normal course of feature development so that we can eventually require linting checks to pass in order to merge PRs.

## Prettier Code Formatting

Prettier has been configured in the project as the code formatter of choice, but it is currently conflicting with a number of code formatting ESLint rules. The prettier ESLint plugin disables these rules if it is placed as an inherited configuration at the end of the `extends` array, so I have made this change. The prettier ESLint plugin also enables Prettier as an ESLint rule, so that code formatting issues become errors. This should be okay, since Prettier fixes them for you.

[M2-4835]: https://mindlogger.atlassian.net/browse/M2-4835?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by farmerpaul — APPROVED** [review]
I'm very happy with these changes, thanks @sultanofcardio. I just left one minor comment about max line length; hoping we can all align on having consistency here (at least between the FE apps, if not the BE).

File: .prettierrc.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-  bracketSameLine: true,
+  arrowParens: 'always',
+  bracketSameLine: false,
   bracketSpacing: true,
-  singleQuote: false,
+  singleQuote: true,
+  jsxSingleQuote: false,
   trailingComma: 'all',
+  semi: true,
   printWidth: 120,
```
[the remark this answers is not in the document]
Comment (sultanofcardio) [reply .prettierrc.js:9]: Thanks for pointing this out Paul. I've been thinking about this and the fact that [Prettier actually recommends](https://prettier.io/docs/en/options.html#print-width) a `printWidth` of 80 for better readability. So I've left this as-is to see if I'd get any feedback on it from the other PRs before making a decision. I might end up going in this direction as you've suggested.

**Review by mbanting — APPROVED** [review]
My review only focused on `.eslintrc.js` and `.prettierrc.js` files. Briefly looking at the rest of the changes they're a result of the changes to these aforementioned files. Thanks for putting this together and incorporating these updated rules. 

Is it possible to hold off on merging until Wed Nov 21? We've been requested to code freeze the `dev2.5` branch to accomodate it's release stabilization. I realize this delay will significantly increase the chance of code conflicts. Maybe this is the first PR that gets merged as soon as the code freeze is lifted.

**Comment by sultanofcardio** [comment]
> Is it possible to hold off on merging until Wed Nov 21

Sure thing @mbanting. I held off https://github.com/ChildMindInstitute/mindlogger-admin/pull/1472 for the same reason 👍

**Comment by mbanting** [comment]
> > Is it possible to hold off on merging until Wed Nov 21
> 
> Sure thing @mbanting. I held off [ChildMindInstitute/mindlogger-admin#1472](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1472) for the same reason 👍

Actually feel free to merge that one. We've only been ask to code freeze the `dev2.5` branch for this repo, `mindlogger-web-refactor`. Thk you!

**Review by moiskillnadne — APPROVED** [review]
In my humble opinion, the semicolon in JS code makes it more overloaded for the eyes. Probably it is just my habit. 

But to make the codebase consistent between repositories we should do it. It looks good to me.

Resolve conflicts please and merge it.

**Pull request merged**